src/legacy/plugin_discovery/types.ts

```diff
@@ -72,8 +72,10 @@ export interface LegacyPluginOptions {
     visTypes: string[];
     embeddableActions?: string[];
     embeddableFactories?: string[];
+    interpreter?: string[];
+    uiSettingDefaults?: Record<string, any>;
   }>;
-  uiCapabilities?: Capabilities;
+  uiCapabilities?: Capabilities | (() => Record<string, any>);
```

Comment:
Not sure that we need this change

Reply:
It's needed. Else `uiCapabilities` throws an error in `index.ts`

Reply:
Could we instead type this as `Capabilities | () => Capabilities`?

The typing here is fine for `timelion` specifically, but I think it would future-proof us for the rest of the migration process if we kept the original `Capabilities` types rather than allowing `Record<string, any>`

Reply:
Changed as a _Partial_. Else some properties should be defined despite their absence.